File: InternalLLMBasedInspectionTool.Frontend/src/app/models/code-attachments/save-code-attachment-request.model.ts
import { CodeLanguage } from './code-attachment.model';

export interface SaveCodeAttachmentRequest {
    userId: string;
    name: string;
    mimeType: string;
    codeLanguage: CodeLanguage;
    originalData: string;
    editedData: string;
    diffData: string;
}

File: InternalLLMBasedInspectionTool.Backend/InternalLLMBasedInspectionTool.WebApi/Models/CodeAttachments/SaveCodeAttachmentRequestModel.cs
using InternalLLMBasedInspectionTool.Domain.CodeAttachments;

namespace InternalLLMBasedInspectionTool.WebApi.Models.CodeAttachments;

public record SaveCodeAttachmentRequestModel {
    public required Guid UserId { get; init; }
    public required string Name { get; init; }
    public required string MimeType { get; init; }
    public required CodeLanguage CodeLanguage { get; init; }
    public required string OriginalData { get; init; }
    public required string EditedData { get; init; }
    public required string DiffData { get; init; }
}